docs/learn/fundamentals/fees-resource-limits-metering.mdx

```diff
@@ -0,0 +1,212 @@
+
+## Surge and dynamic pricing
+
+### Surge pricing
+
+The network can enter surge pricing mode under two circumstances: 1. when the number of operations submitted to a ledger exceeds the network capacity (on Mainnet: 1,000 operations for transactions that do not execute smart contracts), or 2. if there is competition between smart contract transactions for a particular resource (instructions, ledger entry accesses (reads and writes), ledger IO (bytes read and bytes written), and the total size of transactions to be applied). During this time, the network uses market dynamics to decide which transactions to include in the ledger. Transactions that offer a higher maximum base fee bid make it to the ledger first.
+
+During surge pricing mode, transactions are sorted based on their inclusion fee amount, and the user pays the minimum inclusion fee in their transaction set. For example, if there are five transactions with respective inclusion fees of 2, 3, 4, 4, and 5 XLM, and only four of them make it to the ledger, then all included transactions pay the inclusion fee of 3 XLM. If all five transactions made it to the ledger (which would take it out of surge pricing mode), each would pay the minimum inclusion fee of 100 stroops (.0001 XLM).
+
+If there are multiple transactions offering the same inclusion fee, but they cannot all fit into the ledger, transactions are picked randomly so that the total operations for the entire set don’t exceed 1,000. The rest of the transactions are pushed to the next ledger or discarded if they’ve been waiting for too long. If your transaction is discarded, Horizon will return a timeout error.
+
+:::note
+
+It is recommended to apply [ledger bounds](./stellar-data-structures/operations-and-transactions#ledger-bounds) or [time bounds](./stellar-data-structures/operations-and-transactions#time-bounds) to transactions — either your transaction makes it to the ledger or fails, depending on your time and/or ledger parameters.
+
+:::
+
+**You are more likely to pay a higher inclusion fee when submitting smart contract transactions.** Smart contract transactions have tighter ledger limits than transactions that don’t interact with smart contracts and will therefore experience surge pricing more often. You are more likely to pay your maximum inclusion fee bid or, at least, the minimum inclusion fee bid in your transaction set. So, you must plan your fee bidding strategy accordingly.
+
+### Dynamic pricing for storage
+
+Stellar’s storage database size is determined by two forces: the rate of addition (writes) and the rate of deletions (evictions). Stellar has set a ledger growth threshold to some constant (the `BucketListTargetSizeBytes` network parameter, implemented to prevent explosive state growth and subject to change based on validator vote). Because there is a fixed capacity, write fees are based on the ledger size and can alter dynamically based on that size.
```

Comment:
```suggestion
Stellar’s storage database size is determined by two forces: the rate of additions (writes) and the rate of deletions (evictions). Stellar has set a ledger growth threshold to a constant value (the `BucketListTargetSizeBytes` network parameter, implemented to prevent explosive state growth and subject to change based on validator vote). Because there is a fixed capacity, write fees are based on the ledger size and can alter dynamically based on that size.
```